### PR TITLE
Support MFG creation on sampling gpus for cugraph dgl

### PR DESCRIPTION
**Pull request by VibhuJawa**
This PR builds on top of https://github.com/rapidsai/cugraph/pull/3734

Merge after https://github.com/rapidsai/cugraph/pull/3734

Todo tests for:  
- [x] _get_renumber_map
- [x]  _get_tensor_d_from_sampled_df
- [x] create_homogeneous_sampled_graphs_from_dataframe
- [x] pytests pass
- [x] dataset_from_disk_cudf.ipynb (`obgn_products_sampling/`)
- [x] Verify training benchmark

### CODE REVIEW

File: python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py [hunk 1]
```diff
@@ -29,53 +29,114 @@ def cast_to_tensor(ser: cudf.Series):
...
+    return offset_indices
+
+
+def _get_renumber_map(df):
+    map = df["map"]
+    df.drop(columns=["map"], inplace=True)
+
+    map_starting_offset = map.iloc[0]
+    renumber_map = map[map_starting_offset:].dropna().reset_index(drop=True)
+    offsets_per_batch = map[1:map_starting_offset].reset_index(drop=True)
```
Comment (alexbarghi-nv) [inline python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py:67]: Is this correct?  You're separating the offsets from the map, and reindexing the map, but you're not subtracting off `map_starting_offset` from the elements in `offsets_per_batch`.

Comment (alexbarghi-nv) [reply python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py:67]: I think you need to change this to something like:
```
offsets_per_batch = map[1:map_starting_offset].reset_index(drop=True) - map_starting_offset
```

Comment (VibhuJawa) [reply python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py]: I think the variable name was wrong . Changed it to  `renumber_map_batch_indices ` .  

I think the implementation is correct but let me test more.

Comment (alexbarghi-nv) [reply python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py]:
````
map
4       # start of batch 0
10     # start of batch 1
15     # start of batch 2
17     # end of batch 2
6       # first renumber map starts here, at index 4
5
2
3
8
1
1       # second renumber map starts here, at index 10
8
2
11
14
8       # third renumber map starts here, at index 15
9
         # third renumber map ends here, at the length (17)
```

So, to get the map for each batch, you do:
```
for b in range(num_batches):
     indices_batch_b = map.iloc[[b, b+1]]
     start_ix_map = indices_batch_b.iloc[0]
     end_ix_map = indices_batch_b.iloc[1]
     map_batch_b = map.iloc[start_ix_map:end_ix_map]

     # do the rest of the processing for the current batch
```

Comment (alexbarghi-nv) [reply python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py]: This seems to be resolved with your latest change 👍

**Review by alexbarghi-nv — APPROVED** [review]
👍

File: python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py (hunk 1 above)
Comment (VibhuJawa) [reply python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py]: Thanks for catching this @alexbarghi-nv .  I appreciate the detailed explanation. 


I added a test to catch this behavior. 

https://github.com/rapidsai/cugraph/blob/4c2a42de21edb472156f96423dac51d8a1457e23/python/cugraph-dgl/tests/test_utils.py#L68-L75




@alexbarghi-nv , Can you verify that the map in below dataframe  is valid , please. 


https://github.com/rapidsai/cugraph/blob/4c2a42de21edb472156f96423dac51d8a1457e23/python/cugraph-dgl/tests/test_utils.py#L35-L43

Comment (alexbarghi-nv) [reply python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py]: LGTM

**Comment by VibhuJawa** [comment]
@alexbarghi-nv / @rlratzel ,  This is good to go from my end now.

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Pull request merged**